### ISSUE TITLE
Questions regarding datasets

### ISSUE THREAD

**Issue by young-su-ko** [kundajelab/ImmuneCLIP#1]
Congrats on the acceptance to RECOMB!

I have a few questions about the datasets used after reading the pre-print available on Biorxiv.

First, how are the negative epitope/TCR interactions generated for the training data?

Second, for the TULIP and STAPLER test sets, it is mentioned certain datapoints are excluded for a fair comparison. Can you provide the list of datapoints removed, or the filtered version of the test set that you used? What percent of the test set is being removed in this step?

And lastly, will the exact 70/15/15 dataset splits be deposited and made publicly available? 

Thank you,
Young Su Ko

**Comment by lordim** [comment]
Thanks Young Su! Regarding your questions,

1. We do not have _explicit_ negative interactions used during training. Because it is a CLIP-style training framework, at any given batch of epitope-TCR pairs, the off-diagonal entries in the dot product matrix are implicitly considered as negative examples by the CLIP objective.

2. For the dataset filtering, we excluded examples for any epitopes that had been seen by the other models during training but were not in our model’s training set, and vice versa. For example, if epitope "DIVLQVTWS" was not part of ImmuneCLIP's train set but was part of the test set for TULIP (or STAPLER), we excluded that from the test set. Also, if epitope "MFPTYQVSS" was NOT seen during training by TULIP (or STAPLER) but is in the test set, we excluded them as well (because we're only evaluating the performance on _seen_ epitopes)

3. The exact 70/15/15 dataset split should be reproducible with the default random_seed that we used in the training arguments.

Let me know if you have more questions.

**Comment by young-su-ko** [comment]
Thanks for the thorough explanation! Will you be presenting in person at RECOMB? I would love to chat more about ImmuneCLIP and your other awesome AbMAP work!

**Comment by lordim** [comment]
Yes, I'll be presenting in person at RECOMB. I'll see you there!
